internal/metamorph/processor.go

```diff
@@ -857,3 +888,25 @@ func callbackExists(callback store.Callback, data *store.Data) bool {
 	}
 	return false
 }
+
+func (p *Processor) saveTxToCache(hash *chainhash.Hash) error {
+	return p.cacheStore.MapSet(CacheRegisteredTxsHash, hash.String(), []byte("1"))
+}
+
+func (p *Processor) txFoundInCache(hash *chainhash.Hash) (found bool) {
+	value, err := p.cacheStore.MapGet(CacheRegisteredTxsHash, hash.String())
+	if err != nil {
+		p.logger.Error("count not get the transaction from hash", slog.String("hash", hash.String()), slog.String("err", err.Error()))
+		// respond as if the tx is found just in case this transaction is registered
+		return true
+	}
+
+	return value != nil
+}
+
+func (p *Processor) delTxFromCache(hash *chainhash.Hash) {
+	err := p.cacheStore.MapDel(CacheRegisteredTxsHash, hash.String())
+	if err != nil {
+		p.logger.Error("unable to delete transaction from cache", slog.String("hash", hash.String()), slog.String("err", err.Error()))
```

Comment:
IMO this should return the error, not log it as in `saveTxToCache`

Reply:
Done